--- 16.2.14/mixin-css/all-component-typographies
+++ candidate/mixin-css/all-component-typographies
@@ -8,2 +8,28 @@
 
+.sass-api-probe {
+  --mat-badge-container-shape: 50%;
+  --mat-badge-container-size: unset;
+  --mat-badge-small-size-container-size: unset;
+  --mat-badge-large-size-container-size: unset;
+  --mat-badge-legacy-container-size: 22px;
+  --mat-badge-legacy-small-size-container-size: 16px;
+  --mat-badge-legacy-large-size-container-size: 28px;
+  --mat-badge-container-offset: -11px 0;
+  --mat-badge-small-size-container-offset: -8px 0;
+  --mat-badge-large-size-container-offset: -14px 0;
+  --mat-badge-container-overlap-offset: -11px;
+  --mat-badge-small-size-container-overlap-offset: -8px;
+  --mat-badge-large-size-container-overlap-offset: -14px;
+  --mat-badge-container-padding: 0;
+  --mat-badge-small-size-container-padding: 0;
+  --mat-badge-large-size-container-padding: 0;
+  --mat-badge-text-font: Roboto, sans-serif;
+  --mat-badge-line-height: 22px;
+  --mat-badge-text-size: 12px;
+  --mat-badge-text-weight: 600;
+  --mat-badge-small-size-text-size: 9px;
+  --mat-badge-small-size-line-height: 16px;
+  --mat-badge-large-size-text-size: 24px;
+  --mat-badge-large-size-line-height: 28px;
+}
 .sass-api-probe .mat-h1,
@@ -151,2 +177,10 @@
 .sass-api-probe {
+  --mat-bottom-sheet-container-shape: 4px;
+  --mat-bottom-sheet-container-text-font: Roboto, sans-serif;
+  --mat-bottom-sheet-container-text-line-height: 20px;
+  --mat-bottom-sheet-container-text-size: 14px;
+  --mat-bottom-sheet-container-text-tracking: 0.0178571429em;
+  --mat-bottom-sheet-container-text-weight: 400;
+}
+.sass-api-probe {
   --mat-legacy-button-toggle-text-font: Roboto, sans-serif;
@@ -155,2 +189,21 @@
 
+.sass-api-probe {
+  --mat-button-toggle-focus-state-layer-opacity: 0.12;
+  --mat-button-toggle-hover-state-layer-opacity: 0.04;
+  --mat-button-toggle-legacy-focus-state-layer-opacity: 1;
+  --mat-button-toggle-legacy-height: 36px;
+  --mat-button-toggle-legacy-shape: 2px;
+  --mat-button-toggle-shape: 4px;
+  --mat-button-toggle-label-text-font: Roboto, sans-serif;
+  --mat-button-toggle-label-text-line-height: 24px;
+  --mat-button-toggle-label-text-size: 16px;
+  --mat-button-toggle-label-text-tracking: 0.03125em;
+  --mat-button-toggle-label-text-weight: 400;
+  --mat-button-toggle-legacy-label-text-font: Roboto, sans-serif;
+  --mat-button-toggle-legacy-label-text-line-height: 24px;
+  --mat-button-toggle-legacy-label-text-size: 16px;
+  --mat-button-toggle-legacy-label-text-tracking: 0.03125em;
+  --mat-button-toggle-legacy-label-text-weight: 400;
+  --mat-divider-width: 1px;
+}
 .sass-api-probe {
@@ -166,2 +219,16 @@
 
+.sass-api-probe {
+  --mat-datepicker-calendar-container-shape: 4px;
+  --mat-datepicker-calendar-container-touch-shape: 4px;
+  --mat-datepicker-calendar-container-elevation-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
+  --mat-datepicker-calendar-container-touch-elevation-shadow: 0px 11px 15px -7px rgba(0, 0, 0, 0.2), 0px 24px 38px 3px rgba(0, 0, 0, 0.14), 0px 9px 46px 8px rgba(0, 0, 0, 0.12);
+  --mat-datepicker-calendar-text-font: Roboto, sans-serif;
+  --mat-datepicker-calendar-text-size: 13px;
+  --mat-datepicker-calendar-body-label-text-size: 14px;
+  --mat-datepicker-calendar-body-label-text-weight: 500;
+  --mat-datepicker-calendar-period-button-text-size: 14px;
+  --mat-datepicker-calendar-period-button-text-weight: 500;
+  --mat-datepicker-calendar-header-text-size: 11px;
+  --mat-datepicker-calendar-header-text-weight: 400;
+}
 .sass-api-probe {
@@ -180,2 +247,15 @@
 .sass-api-probe {
+  --mat-expansion-container-shape: 4px;
+  --mat-expansion-legacy-header-indicator-display: inline-block;
+  --mat-expansion-header-indicator-display: none;
+  --mat-expansion-header-text-font: Roboto, sans-serif;
+  --mat-expansion-header-text-size: 14px;
+  --mat-expansion-header-text-weight: 500;
+  --mat-expansion-header-text-line-height: inherit;
+  --mat-expansion-header-text-tracking: inherit;
+  --mat-expansion-container-text-font: Roboto, sans-serif;
+  --mat-expansion-container-text-line-height: 20px;
+  --mat-expansion-container-text-size: 14px;
+  --mat-expansion-container-text-tracking: 0.0178571429em;
+  --mat-expansion-container-text-weight: 400;
   --mat-grid-list-tile-header-primary-text-size: 14px;
@@ -184,4 +264,5 @@
   --mat-grid-list-tile-footer-secondary-text-size: 12px;
-}
-
+  --mat-sidenav-container-shape: 0;
+  --mat-sidenav-container-width: auto;
+}
 .sass-api-probe {
@@ -197,2 +278,11 @@
 .sass-api-probe {
+  --mat-stepper-header-focus-state-layer-shape: 4px;
+  --mat-stepper-header-hover-state-layer-shape: 4px;
+  --mat-stepper-container-text-font: Roboto, sans-serif;
+  --mat-stepper-header-label-text-font: Roboto, sans-serif;
+  --mat-stepper-header-label-text-size: 14px;
+  --mat-stepper-header-label-text-weight: 400;
+  --mat-stepper-header-error-state-label-text-size: 16px;
+  --mat-stepper-header-selected-state-label-text-size: 16px;
+  --mat-stepper-header-selected-state-label-text-weight: 400;
   --mat-toolbar-title-text-font: Roboto, sans-serif;
@@ -203,3 +293,2 @@
 }
-
 .sass-api-probe .mat-tree {
@@ -211,2 +300,7 @@
   font-size: 14px;
+}
+.sass-api-probe {
+  --mat-tree-node-text-font: Roboto, sans-serif;
+  --mat-tree-node-text-size: 14px;
+  --mat-tree-node-text-weight: 400;
 }
@@ -415,2 +509,8 @@
 }
+.sass-api-probe .mat-mdc-button,
+.sass-api-probe .mat-mdc-raised-button,
+.sass-api-probe .mat-mdc-unelevated-button,
+.sass-api-probe .mat-mdc-outlined-button {
+  line-height: inherit;
+}
 .sass-api-probe .mdc-fab--extended {
