--- 16.2.14/mixin-css/button-toggle-density
+++ candidate/mixin-css/button-toggle-density
@@ -3 +3,11 @@
 }
+
+.sass-api-probe {
+  --mat-button-toggle-focus-state-layer-opacity: 0.12;
+  --mat-button-toggle-hover-state-layer-opacity: 0.04;
+  --mat-button-toggle-legacy-focus-state-layer-opacity: 1;
+  --mat-button-toggle-legacy-height: 36px;
+  --mat-button-toggle-legacy-shape: 2px;
+  --mat-button-toggle-shape: 4px;
+  --mat-button-toggle-height: 48px;
+}
