--- 16.2.14/mixin-css/private-strong-focus-indicators-structural-styling
+++ candidate/mixin-css/private-strong-focus-indicators-structural-styling
@@ -17,10 +17,11 @@
   content: "";
 }
 
-.cdk-high-contrast-active {
-  --mat-focus-indicator-display: block;
+@media (forced-colors: active) {
+  html {
+    --mat-focus-indicator-display: block;
+  }
 }
-
 .mat-mdc-focus-indicator {
   position: relative;
 }
@@ -40,6 +41,8 @@
   content: "";
 }
 
-.cdk-high-contrast-active {
-  --mat-mdc-focus-indicator-display: block;
+@media (forced-colors: active) {
+  html {
+    --mat-mdc-focus-indicator-display: block;
+  }
 }