--- 16.2.14/mixin-css/stepper-typography
+++ candidate/mixin-css/stepper-typography
@@ -9 +9,13 @@
 }
+
+.sass-api-probe {
+  --mat-stepper-header-focus-state-layer-shape: 4px;
+  --mat-stepper-header-hover-state-layer-shape: 4px;
+  --mat-stepper-container-text-font: Roboto, sans-serif;
+  --mat-stepper-header-label-text-font: Roboto, sans-serif;
+  --mat-stepper-header-label-text-size: 14px;
+  --mat-stepper-header-label-text-weight: 400;
+  --mat-stepper-header-error-state-label-text-size: 16px;
+  --mat-stepper-header-selected-state-label-text-size: 16px;
+  --mat-stepper-header-selected-state-label-text-weight: 400;
+}
