--- 16.2.14/mixin-css/toolbar-theme
+++ candidate/mixin-css/toolbar-theme
@@ -29 +29,25 @@
 }
+
+.sass-api-probe {
+  --mat-toolbar-container-background-color: white;
+  --mat-toolbar-container-text-color: rgba(0, 0, 0, 0.87);
+  --mat-toolbar-title-text-font: Roboto, sans-serif;
+  --mat-toolbar-title-text-line-height: 32px;
+  --mat-toolbar-title-text-size: 20px;
+  --mat-toolbar-title-text-tracking: 0.0125em;
+  --mat-toolbar-title-text-weight: 500;
+  --mat-toolbar-standard-height: 64px;
+  --mat-toolbar-mobile-height: 56px;
+}
+.sass-api-probe .mat-toolbar.mat-primary {
+  --mat-toolbar-container-background-color: #3f51b5;
+  --mat-toolbar-container-text-color: white;
+}
+.sass-api-probe .mat-toolbar.mat-accent {
+  --mat-toolbar-container-background-color: #ff4081;
+  --mat-toolbar-container-text-color: white;
+}
+.sass-api-probe .mat-toolbar.mat-warn {
+  --mat-toolbar-container-background-color: #f44336;
+  --mat-toolbar-container-text-color: white;
+}
